### DISCUSSION TITLE
If page.Target is deprecated, how do we get the TargetId?

### DISCUSSION THREAD

**Discussion by blackcid** [hardkoded/puppeteer-sharp#2830] [Q&A] [answered]
If page.Target is deprecated, how do we get the TargetId?

Thanks in advance

**Comment by kblok** [comment, marked as answer]
@blackcid TargetId is deprecated because it's tied to CDP and we're preparing for Bidi.
If you need the target ID, you can cast your `IPage` to `CdpPage` and access its target. Something like `((CdpPage)page).Target.TargetId`.

**Reply by danya02** [reply]
Is there some equivalent feature in Bidi to figure out the identity of a page? From what I can tell, the identity of C# objects don't relate to this.

**Reply by kblok** [reply]
You should still be able to use Page.MainFrame.Id. Would that help?
